custom_components/waste_collection_schedule/waste_collection_schedule/source/egn_abfallkalender_de.py

```diff
@@ -0,0 +1,91 @@
+import datetime
+import urllib
+import logging
+from typing import Dict
+
+import requests
+from bs4 import BeautifulSoup
+from requests.sessions import dispatch_hook
+from waste_collection_schedule import Collection  # type: ignore[attr-defined]
+
+TITLE = "EGN Abfallkalender"
+DESCRIPTION = "Source for EGN Abfallkalender"
+URL = "https://www.egn-abfallkalender.de/kalender"
+TEST_CASES: Dict[str, Dict[str, str]] = {
+    "Grevenbroich": {"city": "Grevenbroich" , "district": "Noithausen", "street": "Von-Immelhausen-Straße", 
+        "housenumber": "12", "days": 7},
+    "Dormagen": {"city": "Dormagen" , "district": "Hackenbroich", "street": "Aggerstraße", 
+        "housenumber": "2", "days": 20},
+    "Grefrath": {"city": "Grefrath" , "district": "Grefrath", "street": "An Haus Bruch", 
+        "housenumber": "18", "days": 8}
+    
+}
+
+HEADERS = {
+    "content-type": "application/x-www-form-urlencoded; charset=UTF-8",
+    "x-requested-with": "XMLHttpRequest",
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+class Source:
+    def __init__(self, city, district, street, housenumber, days=20):
```

Comment:
Could you remove the `days` option and just return the entire period? Filtering is a functionality of the framework and shall not be done by a source.